### PR TITLE
SONARJS-513 NonEmptyCaseWithoutBreak: Fix FP with block

### PR DESCRIPTION
**Pull request by pynicolas**

### CODE REVIEW

File: javascript-checks/src/main/java/org/sonar/javascript/checks/NonEmptyCaseWithoutBreakCheck.java [hunk 1]
```diff
@@ -51,12 +53,22 @@ public void visitSwitchStatement(SwitchStatementTree tree) {
     cases.remove(cases.size() - 1);
     for (SwitchClauseTree switchClauseTree : cases){
       List<StatementTree> statements = switchClauseTree.statements();
-      if (!statements.isEmpty()
-        && !Iterables.getLast(statements).is(Kind.BREAK_STATEMENT, Kind.RETURN_STATEMENT, Kind.THROW_STATEMENT, Kind.CONTINUE_STATEMENT)) {
+      if (!statements.isEmpty() && !endsWithJump(statements)) {
```
Comment (vilchik-elena) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/NonEmptyCaseWithoutBreakCheck.java]: @pynicolas you can drop this `!statements.isEmpty()` as this condition presents in `endsWithJump ()`

File: javascript-checks/src/test/resources/checks/nonEmptyCaseWithoutBreak.js [hunk 2]
```diff
@@ -20,3 +20,16 @@ switch (param) {
   case 0: // OK
     doSomethingElse();
 }
+
+switch (param) {
+  case 0: // OK
+    doSomething(); break;
+  case 1: // OK
+    { break; }
```
Comment (vilchik-elena) [inline javascript-checks/src/test/resources/checks/nonEmptyCaseWithoutBreak.js]: @pynicolas i would add couple more test-cases, like `{ continue; }`, `{ { break } }`

File: javascript-checks/src/main/java/org/sonar/javascript/checks/NonEmptyCaseWithoutBreakCheck.java (hunk 1 above)
Comment (pynicolas) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/NonEmptyCaseWithoutBreakCheck.java]: @vilchik-elena I can't drop `!statements.isEmpty()`: there's a `!` before `endsWithJump(statements)`

Comment (vilchik-elena) [inline javascript-checks/src/main/java/org/sonar/javascript/checks/NonEmptyCaseWithoutBreakCheck.java]: @pynicolas indeed, sorry

**Pull request merged**